cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/FailureCollector.java

```
@@ -38,13 +38,17 @@
    * @param correctiveAction corrective action
    * @return a validation failure
    */
-  ValidationFailure addFailure(String message, @Nullable String correctiveAction);
+  default ValidationFailure addFailure(String message, @Nullable String correctiveAction) {
+    return new ValidationFailure(message, correctiveAction);
```

i think its better to throw unsupportedoperation exception